pandas/core/frame.py

```diff
@@ -4847,8 +4847,13 @@ def _maybe_casted_values(index, labels=None):
                     if issubclass(values_type, DatetimeLikeArray):
                         values = values._data  # TODO: can we de-kludge yet?
 
-                    if mask.any() and isinstance(values, np.ndarray):
-                        values, _ = maybe_upcast_putmask(values, mask, np.nan)
+                    print(mask)
+                    if mask.any():
+                        if isinstance(values, np.ndarray):
+                            values, _ = maybe_upcast_putmask(values, mask, np.nan)
+                        else:
```

Comment:
@jreback added this since you last looked - we handle `nan` in `maybe_upcast_mask` if values is an `ndarray` else fill with `np.nan`